baseapp/baseapp.go

```diff
@@ -534,53 +538,74 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (result sdk
 		ctx = ctx.WithMultiStore(msCache)
 	}
 
-	finalResult := sdk.Result{}
+	// accumulate results
 	var logs []string
```

Comment:
Can you do `make([]string, 0, len(msgs)`, saves overhead due to expanding the array.